### PR TITLE
Update to add memgraph upload and restore script

### PR DESCRIPTION
**Pull request by wfy1997**

### CODE REVIEW

File: data_loader.py [hunk 1]
```diff
@@ -264,12 +265,18 @@ def load(self, file_list, cheat_mode, dry_run, loading_mode, wipe_db, max_violat
...
+                backup_name = datetime.datetime.today().strftime(DATETIME_FORMAT)
+                restore_cmd = backup_neo4j(backup_folder, backup_name, host, self.log)
+                if not restore_cmd:
+                    self.log.error('Backup Neo4j failed, abort loading!')
+                    sys.exit(1)
+            elif self.database_type == "memgraph":
+                backup_name = backup_memgraph(backup_folder, self.memgraph_snapshot_dir, self.log)
+                if not backup_name:
+                    self.log.error('Backup Memgraph failed, abort loading!')
+                    sys.exit(1)
```
Comment (n2iw) [inline data_loader.py:279]: Please provide an error message if database_type is neither.

File: data_loader.py [hunk 2]
```diff
@@ -264,12 +265,18 @@ def load(self, file_list, cheat_mode, dry_run, loading_mode, wipe_db, max_violat
...
-                self.log.error('Backup Neo4j failed, abort loading!')
-                sys.exit(1)
+            if self.database_type == "neo4j":
+                backup_name = datetime.datetime.today().strftime(DATETIME_FORMAT)
+                restore_cmd = backup_neo4j(backup_folder, backup_name, host, self.log)
+                if not restore_cmd:
+                    self.log.error('Backup Neo4j failed, abort loading!')
+                    sys.exit(1)
+            elif self.database_type == "memgraph":
+                backup_name = backup_memgraph(backup_folder, self.memgraph_snapshot_dir, self.log)
```
Comment (n2iw) [inline data_loader.py:276]: Instead of having a separate **memgraph_snapshot_dir** can we use the same **backup_folder** used in Neo4j backup?

**Review by n2iw — CHANGES_REQUESTED** [review]

File: data_loader.py (hunk 2 above)
Comment (wfy1997) [reply data_loader.py:276]: it is using the same backup_folder, the **memgraph_snapshot_dir** is for the location where the memgraph database generate the backup file

File: data_loader.py (hunk 1 above)
Comment (wfy1997) [reply data_loader.py:283]: The error message is added

**Review by n2iw — APPROVED** [review]
LGTM

**Pull request merged**